[CORE-14259] Fixes [UX-518] -add maintenance status to cluster health command

alextreichler opened the pull request:
When reviewing a cluster's health, nodes in Maintenance mode can be easy to overlook, leading to difficulty resolving certain issues. This PR adds maintenance node visibility to `rpk cluster health` to help quickly identify if any nodes are in maintenance mode.

Uses `cl.Brokers()` to fetch the broker list and checks each broker's `maintenance_status.draining` field. This avoids any server-side changes while still surfacing maintenance state in a single additional API call. Nodes in maintenance are only displayed when at least one node is draining.

## Backports Required

- [ ] none - not a bug fix
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [x] v26.2.x
- [x] v26.1.x
- [x] v25.3.x

## UX Changes

* `rpk cluster health` now displays a "Nodes in maintenance" line when one or more nodes are in maintenance mode.

## Release Notes

### Improvements

* `rpk cluster health` will now display any nodes that may be in maintenance mode.

alextreichler committed:
rpk: add maintenance node status to cluster health

Use cl.Brokers() to check maintenance status across all nodes
and surface draining nodes in `rpk cluster health` output.

r-vasquez commented on src/go/rpk/pkg/cli/cluster/health.go:
```diff
@@ -130,20 +135,30 @@ Get cluster health information and exit when the cluster is healthy:
 	if len(nodesInRecoveryMode) == 0 {
 		nodesInRecoveryMode = []int{}
 	}
+	var nodesInMaintenance []int
+	for _, b := range brokers {
+		if b.Maintenance != nil && b.Maintenance.Draining {
+			nodesInMaintenance = append(nodesInMaintenance, b.NodeID)
+		}
+	}
+	if len(nodesInMaintenance) == 0 {
+		nodesInMaintenance = []int{}
+	}
```
```suggestion
    nodesInMaintenance := []int{}
	for _, b := range brokers {
		if b.Maintenance != nil && b.Maintenance.Draining {
			nodesInMaintenance = append(nodesInMaintenance, b.NodeID)
		}
	}
```

You could do this too, I guess this is to avoid printing `null` in the json response, right?

r-vasquez reviewed (approved):
LGTM, just one minor non-blocking comment, let me know if you fix it and I'll re-approve.

andrwng committed:
rpk: simplify nodesInMaintenance slice initialization

Initialize nodesInMaintenance as an empty slice up front rather than
appending to a nil slice and normalizing to empty afterwards, keeping
the JSON/YAML output from rendering null. Addresses review feedback.

andrwng commented:
@r-vasquez chatted offline with Alex since he's on PTO. I'm happy to carry this forward. Pushed your suggestion and updated the backports list (presumably the original intention was all previous branches?)

vbotbuildovich commented:
/backport v26.2.x

vbotbuildovich commented:
/backport v26.1.x

vbotbuildovich commented:
/backport v25.3.x
